app/services/image_characterizer_service.rb

```diff
@@ -38,7 +38,11 @@ def extract_attributes(output, filepath)
     json_output.each do |file|
       next unless file['SourceFile'] == filepath
 
-      return file['ImageHeight'], file['ImageWidth']
+      attributes = {}.tap do |metadata|
+        metadata[:height] = file['ImageHeight'] unless file['ImageHeight'].nil?
+        metadata[:width] = file['ImageWidth'] unless file['ImageWidth'].nil?
+      end
+      return attributes.empty? ? nil : attributes
```

Comment:
What would you think about using `attributes.presence` here?

Reply:
presence it is.